contracts/libraries/FVMLib.sol

```diff
@@ -91,8 +91,8 @@ error InvalidBytesLength(uint256 expected, uint256 length); // 0xe19dc95e
  * | bytes[0]                   | 0xAA Instruction code     |
  * | bytes[1]                   | Amount of Instructions    |
  * | bytes[2]                   | ptr[0] := Length of instruction[0]
- * | bytes[2:ptr[0] + 1]        | Data of instruction[0]. Calldata slice does not include end index.   |
- * | bytes[ptr[0] + 1]          | ptr[1] := Length of instruction[1] |
+ * | bytes[3:3 + ptr[0]]        | Data of instruction[0]. Calldata slice does not include end index.   |
```

Comment:
I believe 2 as the starting index for the slice was actually correct. See:

Line 103: `idxPtr` initialized to 2
Line 111: `idxInstructionStart` set to `idxPtr`, so on first execution, it's 2
Line 122: `instruction` is set to a slice starting at `idxInstructionStart` (which was not modified since it was set)
Line 128: `_process` called on a slice of `instruction` starting from index 1

Of course I guess the code doesn't have to strictly match the description here so long as it is conceptually correct. That said, you could also change the code to slice out `instruction` starting from `idxInstructionStart+1` and then not need to slice on line 128, which would match these updated comments and should also save a smidgeon of gas.

Reply:
Good idea, I updated the slicing of the `instruction` variable.